core/extraction/towns_fund_round_three.py

```diff
@@ -856,14 +857,19 @@ def extract_outcomes(df_input: pd.DataFrame, project_lookup: dict, programme_id:
     if invalid_projects := relevant_projects - set(project_lookup.keys()):
         raise ValidationError(
             [
-                vf.InvalidOutcomeProjectFailure(
-                    invalid_project=row["Relevant project(s)"],
+                vf.GenericFailure(
+                    sheet="Outcomes",
                     section="Outcome Indicators (excluding footfall)",
-                    row_indexes=[idx + 2],  # +2 here as caught mid-ingest before post-transformation incrementation
+                    # +2 here as caught mid-ingest before post-transformation incrementation
+                    cell_index=vf.construct_cell_index(
+                        table="Outcome_Data", column="Relevant project(s)", row_index=idx + 2
+                    ),
+                    message=msgs.DROPDOWN,
                 )
-                for idx, row in outcomes_df.loc[outcomes_df["Relevant project(s)"].isin(invalid_projects)].iterrows()
+                for idx, _ in outcomes_df.loc[outcomes_df["Relevant project(s)"].isin(invalid_projects)].iterrows()
```

Comment:
It was rebased shortly before your review - I imagine you didn't have the changes

Reply:
> Failure to select a valid project for a given outcome is a GenericFailure that relays a msgs.DROPDOWN message to the user, but we still have a class called InvalidEnumFailure that functions in a similar way and returns the same message. What qualifies something as a GenericFailure vs any of the other classes of failures?

Fair point - I'll look at this

Reply:
I think the reason it makes sense to use a `GenericFailure` like this is because we know all the details of the Failure when we instantiate it because it's being called from a Towns Fund Round 3/4 specific level - just like the Round 4 specific validations. The `Failure`s produced in `validation.py` all have to go through some translation stage in order to map it back to the original spreadsheet.. I think this is the distinction

Reply:
Alright, so ```GenericFailure``` is effectively a failure we can't neatly slot into another category.